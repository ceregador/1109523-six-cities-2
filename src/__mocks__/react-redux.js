export const connectAdvanced = jest.fn().mockImplementation((component) => component);
const mockConnect = () => connectAdvanced;

export const connect = jest.fn().mockImplementation(mockConnect);
