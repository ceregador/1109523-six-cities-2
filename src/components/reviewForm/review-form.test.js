import React from 'react';
import {connect, connectAdvanced} from 'react-redux';
import ReviewForm from './review-form.jsx';
import RatingStar from '../ratingStar/rating-star.jsx';
import Renderer from 'react-test-renderer';

jest.mock(`../ratingStar/rating-star.jsx`);
jest.mock(`react-redux`);

it(`renders correctly`, () => {
  const tree = Renderer
      .create(
          <ReviewForm
            addReview={() => null}
            activeOfferId={1}
          />)
      .toJSON();

  expect(connect).toHaveBeenCalledTimes(1);
  expect(connect).toHaveBeenCalledWith(expect.any(Function), expect.any(Object));

  const mapStateToProps = connect.mock.calls[0][0];
  const mappedProps = mapStateToProps({});
  expect(mappedProps).toHaveProperty(`activeOfferId`);

  const mapDispatchToProps = connect.mock.calls[0][1];
  expect(mapDispatchToProps).toHaveProperty(`addReview`);

  expect(connectAdvanced).toHaveBeenCalledWith(ReviewForm);

  expect(RatingStar).toHaveBeenCalledTimes(5);
  expect(tree).toMatchSnapshot();
});
