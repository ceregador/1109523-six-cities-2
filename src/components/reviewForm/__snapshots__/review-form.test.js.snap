// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  action="post"
  className="reviews__form form"
>
  <label
    className="reviews__label form__label"
    htmlFor="review"
  >
    Your review
  </label>
  <div
    className="reviews__rating-form form__rating"
  />
  <span
    style={
      Object {
        "color": "red",
      }
    }
  />
  <textarea
    className="reviews__textarea form__textarea"
    id="review"
    maxLength={300}
    minLength={50}
    name="review"
    onChange={[Function]}
    placeholder="Tell how was your stay, what you like and what can be improved"
    required={true}
  />
  <div
    className="reviews__button-wrapper"
  >
    <p
      className="reviews__help"
    >
      To submit review please make sure to set 
      <span
        className="reviews__star"
      >
        rating
      </span>
       and describe your stay with at least 
      <b
        className="reviews__text-amount"
      >
         characters
      </b>
      .
    </p>
    <button
      className="reviews__submit form__submit button"
      onClick={[Function]}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
